solidity/package.json

```diff
@@ -11,57 +11,68 @@
   },
   "devDependencies": {
     "@layerzerolabs/solidity-examples": "^1.1.0",
-    "@nomiclabs/hardhat-ethers": "^2.2.1",
+    "@nomiclabs/hardhat-ethers": "^2.2.3",
     "@nomiclabs/hardhat-waffle": "^2.0.6",
-    "@typechain/ethers-v5": "^10.0.0",
-    "@typechain/hardhat": "^6.0.0",
+    "@typechain/ethers-v5": "^11.1.2",
+    "@typechain/hardhat": "^9.1.0",
     "chai": "^4.3.6",
     "ethereum-waffle": "^4.0.10",
     "ethers": "^5.7.2",
-    "hardhat": "^2.19.0",
+    "hardhat": "^2.22.2",
     "hardhat-gas-reporter": "^1.0.9",
     "prettier": "^2.8.8",
     "prettier-plugin-solidity": "^1.1.3",
     "solhint": "^4.0.0",
     "solhint-plugin-prettier": "^0.0.5",
     "solidity-coverage": "^0.8.3",
     "ts-generator": "^0.1.1",
-    "typechain": "^8.3.2",
-    "typescript": "5.1.6"
+    "ts-node": "^10.8.0",
+    "typechain": "patch:typechain@npm%3A8.3.2#~/.yarn/patches/typechain-npm-8.3.2-b02e27439e.patch",
+    "typescript": "5.3.3"
   },
   "directories": {
     "test": "test"
   },
+  "type": "module",
+  "exports": {
+    ".": "./dist/index.js",
+    "./mailbox": "./dist/contracts/Mailbox.js",
+    "./buildArtifact.json": "./buildArtifact.json",
+    "./contracts": "./contracts"
+  },
+  "types": "./dist/index.d.ts",
   "files": [
     "/buildArtifact.json",
     "/dist",
-    "/contracts",
-    "/interfaces",
-    "/docs"
+    "/contracts"
   ],
+  "engines": {
+    "node": ">=16"
+  },
   "homepage": "https://www.hyperlane.xyz",
+  "repository": "https://github.com/hyperlane-xyz/hyperlane-monorepo",
   "keywords": [
     "Hyperlane",
     "Solidity"
   ],
   "license": "Apache-2.0",
-  "main": "dist/index.js",
-  "repository": "https://github.com/hyperlane-xyz/hyperlane-monorepo",
   "scripts": {
-    "build": "hardhat compile && ./exportBuildArtifact.sh && tsc",
+    "build": "yarn hardhat-esm compile && ./exportBuildArtifact.sh && tsc",
     "lint": "solhint contracts/**/*.sol",
-    "clean": "hardhat clean && rm -rf ./dist ./cache ./types ./coverage",
+    "clean": "yarn hardhat-esm clean && rm -rf ./dist ./cache ./types ./coverage ./out ./forge-cache",
     "coverage": "./coverage.sh",
     "docs": "forge doc",
-    "storage": "./storage.sh",
+    "hardhat-esm": "NODE_OPTIONS='--experimental-loader ts-node/esm/transpile-only --no-warnings=ExperimentalWarning' hardhat --config hardhat.config.cts",
```

Comment:
This long line is necessary to get hardhat's internal use of ts-node work. It comes from their recommendation here: https://github.com/NomicFoundation/hardhat/issues/3385#issuecomment-1841380253